pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -150,17 +169,67 @@ func (state PipelineRunState) IsDone() (isDone bool) {
 // GetNextTasks will return the next ResolvedPipelineRunTasks to execute, which are the ones in the
 // list of candidateTasks which aren't yet indicated in state to be running.
 func (state PipelineRunState) GetNextTasks(candidateTasks map[string]struct{}) []*ResolvedPipelineRunTask {
```

Comment:
this is where the most of the changes happened, for a task before adding it to the queue, check if `runAfter` is specified, check if `runOn` is specified, match the state in case `runOn` is specified. Mark the task as conflicted when `runOn` states does not match with the actual state, etc

Reply:
I commented this already, but instead of marking the TaskRun as conflicted, could we simply not return it here?  Is the "conflicted" status used to avoid having to match the runOn filters in various reconcile cycles?

Doesn't this pose the problem that if a taskrun depended on a conflicted one, with a runOn(failure), it would then get execute, instead of being skipped too?